openedx/adg/lms/applications/models.py

```diff
@@ -295,6 +295,27 @@ def is_education_experience_completed(self):
         return self.is_education_completed and self.is_work_experience_completed \
             and bool(self.background_question) and self.is_reference_added
 
+    @property
+    def date_accepted(self):
+        """
+        Fetch the date the application decision is made
+
+        Returns:
+            datetime: Date when the UserApplication's decision is made
+        """
+        return self.modified
+
+    # pylint: disable=no-member
+    @property
+    def date_submitted(self):
+        """
+        Fetch the date the written application is submitted
+
+        Returns:
+            datetime: Date when the UserApplication is submitted
+        """
+        return self.user.application_hub.submission_date
```

Comment:
will this result in exception if there is no application hub?

Reply:
Can there ever be no application hub? 
I'll rewrite this with a sanity check

Reply:
In order to reach the User Application pages, we check to make sure that the ApplicationHub exists. We do not need a check here